internal/ir/xds.go

```diff
@@ -1551,6 +1552,12 @@ type ActiveHealthCheck struct {
 	TCP *TCPHealthChecker `json:"tcp,omitempty" yaml:"tcp,omitempty"`
 }
 
+func (h *HealthCheck) SetHTTPHostIfAbsent(host string) {
+	if h != nil && h.Active != nil && h.Active.HTTP != nil && h.Active.HTTP.Host == "" {
```

Comment:
do we need a check for `h.Active.HTTP.Host == ""` ?

Reply:
To avoid overriding the user-specified `host` in http health check if defined by users one day.